sssom/io.py

```diff
@@ -228,3 +233,131 @@ def extract_iri(input, prefix_map) -> list:
+#     """Filter mapping file based on prefix and predicates provided.
+
+#     :param input: Input mapping file (tsv)
+#     :param prefix: Prefixes to be retained.
+#     :param predicate: Predicates to be retained.
+#     :return: Filtered MappingSetDataFrame.
+#     """
+#     msdf: MappingSetDataFrame = parse_sssom_table(input)
+#     prefix_map = msdf.prefix_map
+#     df: pd.DataFrame = msdf.df
+#     # Filter prefix_map
+#     filtered_prefix_map = {
+#         k: v for k, v in prefix_map.items() if k in prefix_map.keys() and k in prefix
+#     }
+
+#     filtered_predicates = {
+#         k: v
+#         for k, v in prefix_map.items()
+#         if len([x for x in predicate if str(x).startswith(k)])
+#         > 0  # use re.find instead.
+#     }
+#     filtered_prefix_map.update(filtered_predicates)
+#     filtered_prefix_map = add_built_in_prefixes_to_prefix_map(filtered_prefix_map)
+
+#     # Filter df based on predicates
+#     predicate_filtered_df: pd.DataFrame = df.loc[
+#         df[PREDICATE_ID].apply(lambda x: x in predicate)
+#     ]
+
+#     # Filter df based on prefix_map
+#     prefix_keys = tuple(filtered_prefix_map.keys())
+#     condition_subj = predicate_filtered_df[SUBJECT_ID].apply(
+#         lambda x: str(x).startswith(prefix_keys)
+#     )
+#     condition_obj = predicate_filtered_df[OBJECT_ID].apply(
+#         lambda x: str(x).startswith(prefix_keys)
+#     )
+#     filtered_df = predicate_filtered_df.loc[condition_subj & condition_obj]
+
+#     new_msdf: MappingSetDataFrame = MappingSetDataFrame(
+#         df=filtered_df, prefix_map=filtered_prefix_map, metadata=msdf.metadata
+#     )
+#     return new_msdf
+
+
+def run_sql_query(query: str, inputs: List[str], output: TextIO) -> MappingSetDataFrame:
+    """Run a SQL query over one or more SSSOM files.
+
+    Each of the N inputs is assigned a table name df1, df2, ..., dfN
+
+    Alternatively, the filenames can be used as table names - these are first stemmed
+    E.g. ~/dir/my.sssom.tsv becomes a table called 'my'
+
+    Example:
+        sssom dosql -Q "SELECT * FROM df1 WHERE confidence>0.5 ORDER BY confidence" my.sssom.tsv
+
+    Example:
+        `sssom dosql -Q "SELECT file1.*,file2.object_id AS ext_object_id, file2.object_label AS ext_object_label \
+        FROM file1 INNER JOIN file2 WHERE file1.object_id = file2.subject_id" FROM file1.sssom.tsv file2.sssom.tsv`
+
+    :param query: Query to be executed over a pandas DataFrame (msdf.df).
+    :param inputs: Input files that form the source tables for query.
+    :param output: Output.
+    :return: Filtered MappingSetDataFrame object.
+    """
+    n = 1
+    new_msdf = MappingSetDataFrame()
+    while len(inputs) >= n:
+        fn = inputs[n - 1]
+        msdf = parse_sssom_table(fn)
+        df = msdf.df
+        # df = parse(fn)
+        globals()[f"df{n}"] = df
+        tn = re.sub("[.].*", "", Path(fn).stem).lower()
+        globals()[tn] = df
+        n += 1
+
+    new_df = sqldf(query)
```

Comment:
Hm. I hope you dont mind if I dig one layer deeper, because I really do not understand it.. So the SQL query we pass to pandsql has some kind of FROM clause that knows all by itself to look at globals()[TABLENAME]? In your select clause I see `SELECT * FROM df` - how does know that it should look at input1, input2 etc? Have you tried to run this with two inputs? I am a bit inclined to change the method to only allow a single input..

Reply:
No, `pandasql` is a separate library: https://pypi.org/project/pandasql/